daemons/fenced/fenced_ipc.c

```diff
@@ -67,97 +86,112 @@ fenced_ipc_dispatch(qb_ipcs_connection_t * qbc, void *data, size_t size)
         /* We've read the complete message and there's already a header on
          * the front.  Pass it off for processing.
          */
-        request = pcmk__client_data2xml(c, &id, &flags);
-        g_byte_array_free(c->buffer, TRUE);
-        c->buffer = NULL;
+        msg = pcmk__client_data2xml(client, &id, &flags);
+        g_byte_array_free(client->buffer, TRUE);
+        client->buffer = NULL;
 
     } else {
         /* Some sort of error occurred reassembling the message.  All we can
          * do is clean up, log an error and return.
          */
         crm_err("Error when reading IPC message: %s", pcmk_rc_str(rc));
 
-        if (c->buffer != NULL) {
-            g_byte_array_free(c->buffer, TRUE);
-            c->buffer = NULL;
+        if (client->buffer != NULL) {
+            g_byte_array_free(client->buffer, TRUE);
+            client->buffer = NULL;
         }
 
         return 0;
     }
 
-    if (request == NULL) {
-        pcmk__ipc_send_ack(c, id, flags, PCMK__XE_NACK, NULL, CRM_EX_PROTOCOL);
+    if (msg == NULL) {
+        pcmk__ipc_send_ack(client, id, flags, PCMK__XE_NACK, NULL, CRM_EX_PROTOCOL);
```

Comment:
In the scheduler we also have
```
crm_debug("Unrecognizable IPC data from PID %d", pcmk__client_pid(c));
```

Doesn't matter to me. It's also an ACK in the scheduler. But you and I have discussed before that we don't know how ACK/NACK is used with these IPC messages.